rb_ws/src/buggy/scripts/auton/model_predictive_controller.py

```diff
@@ -538,27 +543,35 @@ def compute_trajectory(self, current_pose: Pose, trajectory: Trajectory, current
         if self.TIME:
             t = time.time()
 
+        steer_limit = min(self.compute_steering_limit(current_speed), self.MAX_STEER)
+        if self.DEBUG:
+            print("max steer:", np.rad2deg(steer_limit))
+
+        # State constraints
+        state_lb = np.array([-np.inf, -np.inf, -np.inf, -steer_limit])  # x, y, theta, steer
+        state_ub = np.array([np.inf, np.inf, np.inf, steer_limit])  # x, y, theta, steer
+
         lb = np.hstack(
             (
-                -self.state_jacobian(reference_trajectory[0, :])
-                @ (self.state(0, 0, 0, 0) - reference_trajectory[0, :]),
+                -self.state_jacobian(reference_trajectory[0, :]) @ (self.state(0, 0, 0, 0) - reference_trajectory[0, :]),
                 np.zeros(self.N_STATES * (self.MPC_HORIZON - 1)),
-                np.tile(self.state_lb, self.MPC_HORIZON) + reference_trajectory.ravel(),
-                np.tile(self.control_lb, self.MPC_HORIZON) + reference_control.ravel(),
+                np.tile(state_lb, self.MPC_HORIZON) - reference_trajectory.ravel(),
+                np.tile(self.control_lb, self.MPC_HORIZON) - reference_control.ravel(),
                 # np.tile(n.T @ p, self.MPC_HORIZON),
             )
         )
         ub = np.hstack(
             (
-                -self.state_jacobian(reference_trajectory[0, :])
-                @ (self.state(0, 0, 0, 0) - reference_trajectory[0, :]),
+                -self.state_jacobian(reference_trajectory[0, :]) @ (self.state(0, 0, 0, 0) - reference_trajectory[0, :]),
                 np.zeros(self.N_STATES * (self.MPC_HORIZON - 1)),
-                np.tile(self.state_ub, self.MPC_HORIZON) + reference_trajectory.ravel(),
-                np.tile(self.control_ub, self.MPC_HORIZON) + reference_control.ravel(),
+                np.tile(state_ub, self.MPC_HORIZON) - reference_trajectory.ravel(),
+                np.tile(self.control_ub, self.MPC_HORIZON) - reference_control.ravel(),
                 # np.tile(np.inf, self.MPC_HORIZON),
             )
         )
 
+        print("bounds", lb[(self.N_STATES * self.MPC_HORIZON):], ub[(self.N_STATES * self.MPC_HORIZON):])
```

Comment:
Clutter Print? Maybe printing bounds each iteration may be unhelpful and just clutter up stdout.